aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -440,24 +440,60 @@ void normal_fill(Tensor& self, const scalar_t mean, const scalar_t std, Generato
   }
 }
 
+std::vector<int64_t> computeStrideForComplex(IntArrayRef oldstride) {
+  auto res = oldstride.vec();
+  for(size_t i = 0; i < res.size(); i++) {
+    res[i] = res[i] * 2;
+  }
+  res.emplace_back(1);
+  return res;
+}
+
+Tensor view_complex_as_float(const Tensor& self) {
+  auto new_sizes = self.sizes().vec();
+  // last dimension will always have two elements containing the real and imag vals
+  new_sizes.emplace_back(2);
+  auto new_strides = computeStrideForComplex(self.strides());
+  if(self.scalar_type() == at::kComplexFloat) {
+    float* data = reinterpret_cast<float*>(self.data_ptr<std::complex<float>>());
+    return at::from_blob(data, new_sizes, ArrayRef<int64_t>(new_strides), dtype(at::kFloat));
+  } else {
```

Comment:
`TORCH_INTERNAL_ASSERT(self.scalar_type() == at::kComplexDouble)`

Reply:
Though I guess the `data_ptr` access will check this too

Reply:
yeah